RusTaxi/Classes/Changing Button/CustomButton.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  ChangingButton.swift
+//  RusTaxi
+//
+//  Created by Ruslan Prozhivin on 12.10.2018.
+//  Copyright © 2018 App's ID. All rights reserved.
+//
+
+import UIKit
+import Material
+
+class CustomButton: Button {
+	static let shared = CustomButton()
```

Comment:
It's absolutely wrong. You need to inherit from Button your CustomClass and after inherit your button instances from your CustomClass. Singleton here is not need here.